cpp/src/parquet/properties.h

```diff
@@ -322,7 +322,49 @@ class PARQUET_EXPORT WriterProperties {
           content_defined_chunking_enabled_(
               properties.content_defined_chunking_enabled()),
           content_defined_chunking_options_(
-              properties.content_defined_chunking_options()) {}
+              properties.content_defined_chunking_options()) {
+      for (const auto& [col_path, col_props] : properties.column_properties_) {
```

Comment:
I just noticed that `WriterProperties` and `WriterProperties::Builder` use different forms to represent column properties. Is it possible to simplify this by using the same form?